src/lib.rs

```diff
@@ -190,9 +190,20 @@ fn replace_dir<P: AsRef<Path>>(old: P, new: P) -> Result<()> {
     let new = new.as_ref();
 
     if old.exists() && old.is_dir() {
+        // sync the staging directory to ensure our changes have been persisted
+        let old_as_file = File::open(old)
+            .chain_err(|| format!("failed to open old dir '{}' for syncing", old.display()))?;
+        old_as_file.sync_all()
+            .chain_err(|| format!("failed to sync old dir '{}'", old.display()))?;
+
         truncate_dir(new)?;
         fs::rename(old, new)
             .chain_err(|| format!("failed to move '{:?}' to '{:?}'", old, new))?;
+
+        let new_as_file = File::open(new)
+            .chain_err(|| format!("failed to open new dir '{}' for syncing", new.display()))?;
+        new_as_file.sync_all()
```

Comment:
`new_as_file` refers to the same directory as `old_as_file`, so this is redundant with the previous `sync_all`.  What should happen here is that the parent directory of `new` should be synced to ensure the rename is persisted.